### PR TITLE
feat: update secp256k1

### PR DESCRIPTION
**Pull request by ko-matsu**
## Overview

- feat: update secp256k1
  - update ecdsa adaptor

## How to use

```bash
```

## Check list

** Person who issued **
- [ ] checked script 
- [x] build successed
- [ ] Linked PullRequest and Issue

** Reviewer **
- [ ] (if necessary) Record the review results of related tickets.

### CODE REVIEW

File: include/cfdcore/cfdcore_ecdsa_adaptor.h [hunk 1]
```diff
@@ -41,24 +59,104 @@ class CFD_CORE_EXPORT AdaptorSignature {
...
    * @return ByteData
    */
   ByteData GetData() const;
 
+  /**
+   * @brief Get the ByteData object
+   *
+   * @return ByteData
+   */
+  ByteData GetFullyData() const;
```
Comment (Tibo-lg) [inline include/cfdcore/cfdcore_ecdsa_adaptor.h]: What is the difference between `GetData` and `GetFullyData`? (Also maybe `GetFullData` would be better)

File: include/cfdcore/cfdcore_ecdsa_adaptor.h [hunk 2]
```diff
@@ -41,24 +59,104 @@ class CFD_CORE_EXPORT AdaptorSignature {
...
+  bool IsValid() const;
+
  private:
   /**
    * @brief The underlying data
    *
    */
   ByteData data_;
+  /**
+   * @brief The fully signature data
```
Comment (Tibo-lg) [inline include/cfdcore/cfdcore_ecdsa_adaptor.h]: fully -> full

File: include/cfdcore/cfdcore_ecdsa_adaptor.h [hunk 3]
```diff
@@ -41,24 +59,104 @@ class CFD_CORE_EXPORT AdaptorSignature {
...
+   */
+  ByteData sig_data_;
 };
 
 /**
  * @brief A proof for an adaptor signature.
  *
+ * @deprecated This class is deprecated due to refactoring.
```
Comment (Tibo-lg) [inline include/cfdcore/cfdcore_ecdsa_adaptor.h]: Is there a reason to keep it?

File: include/cfdcore/cfdcore_ecdsa_adaptor.h [hunk 4]
```diff
@@ -98,6 +208,7 @@ class CFD_CORE_EXPORT AdaptorProof {
 /**
  * @brief A pair of an adaptor signature together with the proof attached to it.
  *
+ * @deprecated This struct is deprecated due to refactoring.
```
Comment (Tibo-lg) [inline include/cfdcore/cfdcore_ecdsa_adaptor.h]: Same is it useful to keep it?

File: include/cfdcore/cfdcore_ecdsa_adaptor.h [hunk 5]
```diff
@@ -25,9 +27,25 @@ class CFD_CORE_EXPORT AdaptorSignature {
  public:
   /**
    * @brief Size of an adaptor signature.
+   */
+  static constexpr uint32_t kAdaptorSignatureSize = 162;
+  /**
+   * @brief Size of an adaptor signature minimum data.
+   */
+  static constexpr uint32_t kAdaptorSignatureMinimumSize = 65;
```
Comment (Tibo-lg) [inline include/cfdcore/cfdcore_ecdsa_adaptor.h]: I think better to just support the `162` to make things simpler

**Review by Tibo-lg — COMMENTED** [review]
I think it's better to drop support for the old format, but if you prefer to keep it I don't mind either.

**Comment by k-matsuzawa** [comment]
@Tibo-lg  Thanks!

>I think it's better to drop support for the old format, but if you prefer to keep it I don't mind either.

In that case, cfd-core v0.4.x will support only new formats.
(Older formats will be supported by cfd-core 0.3.x.)

**Pull request merged**